src/test/java/com/imagination/cbs/service/impl/EmailServiceImplTest.java

```diff
@@ -0,0 +1,335 @@
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import javax.mail.MessagingException;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.core.env.Environment;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import com.imagination.cbs.domain.ApprovalStatusDm;
+import com.imagination.cbs.domain.Booking;
+import com.imagination.cbs.domain.BookingRevision;
+import com.imagination.cbs.domain.BookingWorkTask;
+import com.imagination.cbs.domain.Contractor;
+import com.imagination.cbs.domain.ContractorEmployee;
+import com.imagination.cbs.domain.Discipline;
+import com.imagination.cbs.domain.OfficeDm;
+import com.imagination.cbs.domain.ReasonsForRecruiting;
+import com.imagination.cbs.domain.RoleDm;
+import com.imagination.cbs.domain.SupplierTypeDm;
+import com.imagination.cbs.dto.InternalResourceEmailDto;
+import com.imagination.cbs.dto.MailRequest;
+import com.imagination.cbs.repository.ApprovalStatusDmRepository;
+import com.imagination.cbs.security.CBSUser;
+import com.imagination.cbs.service.LoggedInUserService;
+import com.imagination.cbs.util.EmailUtility;
+
+import freemarker.core.ParseException;
+import freemarker.template.Configuration;
+import freemarker.template.MalformedTemplateNameException;
+import freemarker.template.Template;
+import freemarker.template.TemplateNotFoundException;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EmailServiceImplTest {
+	
+	@InjectMocks
+	EmailServiceImpl emailServiceImpl;
+	
+	@Mock
+	private EmailUtility emailUtility;
+
+	@Mock
+	private Configuration config;
+
+	@Mock
+	private LoggedInUserService loggedInUserService;
+
+	@Mock
+	private Environment environment;
+
+	@Mock
+	private ApprovalStatusDmRepository approvalStatusDmRepository;
+
+	private Template template;
+
+	@Before
+	public void init() {
+		template = mock(Template.class);
+	}
+	
+	@Test
+	public void shouldSendEmailForBookingApproval() throws TemplateNotFoundException, MalformedTemplateNameException, ParseException, IOException, MessagingException {
+
+		MailRequest emailRequest = createMailRequest();
+		BookingRevision bookingRevision = createBookingRevision(false);
+		ApprovalStatusDm approvalStatusDm = new ApprovalStatusDm();
+		approvalStatusDm.setApprovalName("Akshay");
+		ReflectionTestUtils.setField(emailServiceImpl, "baseUrl", "https://imaginagtion.com/bookingSummary?bookingId={bookingId}&status={status}");
+		
+		when(config.getTemplate("email.approval_request.ftl")).thenReturn(template);
+		when(approvalStatusDmRepository.findById(201L)).thenReturn(Optional.of(approvalStatusDm));
+		
+		emailServiceImpl.sendEmailForBookingApproval(emailRequest, bookingRevision, "approval_request");
+
+		verify(config).getTemplate("email.approval_request.ftl");
```

Comment:
break verify in multiple test cases.

Reply:
Means all the above line of code will be same and only we need to change verify line in another test case, is it so?

Reply:
yes... you can move the common code to some method or in before method .. every thing should be tested separately in separate test case ... thats why we call it unit testing.

Reply:
I still have a doubt, can we have a quick discussion whenever you are free?